diff --git a/cmake/dev-mode.cmake b/cmake/dev-mode.cmake
new file mode 100644
index 0000000..0011f5c
--- /dev/null
+++ b/cmake/dev-mode.cmake
@@ -0,0 +1,21 @@
+include(cmake/folders.cmake)
+
+include(CTest)
+if(BUILD_TESTING)
+  add_subdirectory(test)
+endif()
+
+option(BUILD_MCSS_DOCS "Build documentation using Doxygen and m.css" OFF)
+if(BUILD_MCSS_DOCS)
+  include(cmake/docs.cmake)
+endif()
+
+option(ENABLE_COVERAGE "Enable coverage support separate from CTest's" OFF)
+if(ENABLE_COVERAGE)
+  include(cmake/coverage.cmake)
+endif()
+
+include(cmake/lint-targets.cmake)
+include(cmake/spell-targets.cmake)
+
+add_folders(Project)
