diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 27f26353..49dc35b3 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -1,11 +1,8 @@
 include(GNUInstallDirs)
 
-add_library(kahypar SHARED libkahypar.cc)
+add_library(kahypar STATIC libkahypar.cc)
 target_link_libraries(kahypar ${Boost_LIBRARIES})
 
-set_target_properties(kahypar PROPERTIES
-    PUBLIC_HEADER ../include/libkahypar.h)
-
 target_include_directories(kahypar PRIVATE ../include)
 
 configure_file(libkahypar.pc.in libkahypar.pc @ONLY)
