diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d80ca4a..160aac5c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -351,10 +351,6 @@ message(STATUS "CMAKE_CXX_FLAGS_RELEASE: ${CMAKE_CXX_FLAGS_RELEASE}")
 message(STATUS "CMAKE_CXX_FLAGS_DEBUG: ${CMAKE_CXX_FLAGS_DEBUG}")
 
 
-if(NOT CMAKE_SKIP_INSTALL_RULES)
-  include(cmake/install-rules.cmake)
-endif()
-
 add_subdirectory(kahypar/application)
 
 if(BUILD_TESTING)
@@ -366,6 +362,10 @@ endif()
 add_subdirectory(tools)
 add_subdirectory(lib)
 
+if(NOT CMAKE_SKIP_INSTALL_RULES)
+  include(cmake/install-rules.cmake)
+endif()
+
 if(KAHYPAR_PYTHON_INTERFACE)
   add_subdirectory(python)
 endif()
