diff --git a/cmake/install-rules.cmake b/cmake/install-rules.cmake
new file mode 100644
index 00000000..c22be3d9
--- /dev/null
+++ b/cmake/install-rules.cmake
@@ -0,0 +1,77 @@
+if(PROJECT_IS_TOP_LEVEL)
+  set(
+      CMAKE_INSTALL_INCLUDEDIR "include/KaHyPar-${PROJECT_VERSION}"
+      CACHE PATH ""
+  )
+endif()
+
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+# find_package(<package>) call for consumers to find this project
+set(package KaHyPar)
+
+install(
+    DIRECTORY
+    include/
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+    COMPONENT KaHyPar_Development
+)
+
+install(
+    TARGETS KaHyPar
+    EXPORT KaHyPar
+    RUNTIME #
+    COMPONENT KaHyPar_Runtime
+    LIBRARY #
+    COMPONENT KaHyPar_Runtime
+    NAMELINK_COMPONENT KaHyPar_Development
+    ARCHIVE #
+    COMPONENT KaHyPar_Development
+    INCLUDES #
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+)
+
+write_basic_package_version_file(
+    "${package}ConfigVersion.cmake"
+    COMPATIBILITY SameMajorVersion
+)
+
+if (VCPKG_TARGET_TRIPLET)
+# Allow package maintainers to freely override the path for the configs
+set(
+    KaHyPar_INSTALL_CMAKEDIR "${CMAKE_INSTALL_PREFIX}/share/${package}"
+    CACHE PATH "CMake package config location relative to the install prefix"
+)
+else ()
+set(
+    KaHyPar_INSTALL_CMAKEDIR "${CMAKE_INSTALL_INCLUDEDIR}/cmake/${package}"
+    CACHE PATH "CMake package config location relative to the install prefix"
+)
+endif ()
+
+mark_as_advanced(KaHyPar_INSTALL_CMAKEDIR)
+
+install(
+    FILES cmake/install-config.cmake
+    DESTINATION "${KaHyPar_INSTALL_CMAKEDIR}"
+    RENAME "${package}Config.cmake"
+    COMPONENT KaHyPar_Development
+)
+
+install(
+    FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
+    DESTINATION "${KaHyPar_INSTALL_CMAKEDIR}"
+    COMPONENT KaHyPar_Development
+)
+
+install(
+    EXPORT KaHyParTargets
+    NAMESPACE KaHyPar::
+    DESTINATION "${KaHyPar_INSTALL_CMAKEDIR}"
+    COMPONENT KaHyPar_Development
+)
+
+if(PROJECT_IS_TOP_LEVEL)
+  include(CPack)
+endif()
