diff --git a/samples/XCSP3PrintCallbacks.h b/samples/XCSP3PrintCallbacks.h
index 665c6b8..22f30be 100644
--- a/samples/XCSP3PrintCallbacks.h
+++ b/samples/XCSP3PrintCallbacks.h
@@ -229,7 +229,7 @@ namespace XCSP3Core {
 
         void buildConstraintCircuit(string id, vector<XVariable *> &list, int startIndex, XVariable *size) override;
 
-        void buildConstraintPrecedence(string id, vector<XVariable *> &list, vector<int> values) override;
+        void buildConstraintPrecedence(string id, vector<XVariable *> &list, vector<int> values, bool covered) override;
 
         void buildConstraintBinPacking(string id, vector<XVariable *> &list, vector<int> &sizes, XCondition &cond) override;
 
@@ -237,6 +237,34 @@ namespace XCSP3Core {
 
         void buildConstraintKnapsack(string id, vector<XVariable *> &list, vector<int> &weights, vector<int> &profits,XCondition weightsCondition, XCondition &profitCondition) override;
 
+        void
+        buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths, vector<int> &heights,
+                                  XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                       vector<XVariable *> &varHeights, XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                       vector<int> &heights, XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                       vector<XVariable *> &heights, XCondition &xc) override;
+
+        void
+        buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths, vector<int> &heights,
+                                  vector<XVariable *> &ends, XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                       vector<XVariable *> &varHeights, vector<XVariable *> &ends,
+                                       XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                       vector<int> &heights, vector<XVariable *> &ends, XCondition &xc) override;
+
+        void buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                       vector<XVariable *> &heights, vector<XVariable *> &ends,
+                                       XCondition &xc) override;
+
         void buildObjectiveMinimizeExpression(string expr) override;
 
         void buildObjectiveMaximizeExpression(string expr) override;
@@ -261,6 +289,9 @@ namespace XCSP3Core {
 
 
         void buildAnnotationDecision(vector<XVariable*> &list) override;
+
+        void buildConstraintCount(string id, vector<Tree *> &trees, vector<int> &values, XCondition &xc) override;
+
         bool canonize;
     };
 
@@ -1037,7 +1068,7 @@ void XCSP3PrintCallbacks::buildConstraintNoOverlap(string, vector<XVariable *> &
 
 // string id, vector<XVariable *> &origins, vector<XVariable *> &lengths, bool zeroIgnored
 void XCSP3PrintCallbacks::buildConstraintNoOverlap(string, vector<XVariable *> &origins, vector<XVariable *> &lengths, bool) {
-    cout << "\n    nooverlap constraint" << endl;
+    cout << "\n    nooverlap constraint (origins and lenghts variables)" << endl;
     cout << "        origins:";
     displayList(origins);
     cout << "        lengths";
@@ -1128,12 +1159,17 @@ void XCSP3PrintCallbacks::buildConstraintCircuit(string, vector<XVariable *> &li
     cout << "        size: " << size->id << endl;
 }
 
-void XCSP3PrintCallbacks::buildConstraintPrecedence(string id, vector<XVariable *> &list, vector<int> values) {
+void XCSP3PrintCallbacks::buildConstraintPrecedence(string id, vector<XVariable *> &list, vector<int> values, bool covered) {
     cout << "\n    Precedence constraint" << endl;
     cout << "        list:";
     displayList(list);
     cout << "        values: ";
     displayList(values);
+    cout << "        covered?  " << covered << endl;
+}
+
+void XCSP3PrintCallbacks::buildConstraintCount(string id, vector<Tree *> &trees, vector<int> &values, XCondition &xc) {
+    cout << "\n    Count constraint with trees and integer values" << endl;
 }
 
 
@@ -1223,4 +1259,114 @@ void XCSP3PrintCallbacks::buildAnnotationDecision(vector<XVariable*> &list) {
 
 }
 
+void XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                                    vector<XVariable *> &varHeights, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(varHeights);
+    cout << "        condition:" << xc << endl;
+}
+
+void
+XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                               vector<int> &heights, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        condition:" << xc << endl;
+}
+
+void
+XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                               vector<XVariable *> &heights, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        condition:" << xc << endl;
+}
+
+void XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                                    vector<int> &heights, vector<XVariable *> &ends, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        ends:";
+    displayList(ends);
+    cout << "        condition:" << xc << endl;
+}
+
+void XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                                    vector<XVariable *> &varHeights, vector<XVariable *> &ends,
+                                                    XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(varHeights);
+    cout << "        ends:";
+    displayList(ends);
+    cout << "        condition:" << xc << endl;
+}
+
+void
+XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                               vector<int> &heights, vector<XVariable *> &ends, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        ends:";
+    displayList(ends);
+    cout << "        condition:" << xc << endl;
+}
+
+void
+XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<XVariable *> &lengths,
+                                               vector<XVariable *> &heights, vector<XVariable *> &ends,
+                                               XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        ends:";
+    displayList(ends);
+    cout << "        condition:" << xc << endl;
+}
+
+void XCSP3PrintCallbacks::buildConstraintCumulative(string id, vector<XVariable *> &origins, vector<int> &lengths,
+                                                    vector<int> &heights, XCondition &xc) {
+    cout << "\n    Cumulative constraint" << endl;
+    cout << "        list:";
+    displayList(origins);
+    cout << "        lengths:";
+    displayList(lengths);
+    cout << "        heights:";
+    displayList(heights);
+    cout << "        condition:" << xc << endl;
+}
+
 #endif //COSOCO_XCSP3PRINTCALLBACKS_H
